From 2f5f9afffef1f7115beb16d10f9d1bd16bf823fd Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 17 May 2019 14:30:34 +0000
Subject: [PATCH 0957/1077] MANTIS 0054637 tools/Enumeration.hpp

convert enumeration value to int to avoid ambiguity in << operator

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1451 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Interfaces/generic/tools/Enumeration.hpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Interfaces/generic/tools/Enumeration.hpp b/SelfServiceCommon/Interfaces/generic/tools/Enumeration.hpp
index 74c6020c..7cff1fcb 100644
--- a/SelfServiceCommon/Interfaces/generic/tools/Enumeration.hpp
+++ b/SelfServiceCommon/Interfaces/generic/tools/Enumeration.hpp
@@ -194,7 +194,9 @@ namespace mipsTools
   template<class E>
   std::ostream& operator<<(std::ostream& out, const enumerationEntry<E>& data)
   {
-    out << data.getCode() << "/" << data.getValue() << "/" << data.getDescription();
+    out << static_cast<int>(data.getCode()) << "/"
+        << data.getValue() << "/"
+        << data.getDescription();
     return out;
   }
 }
-- 
2.41.0.windows.1

